src/snowflake/snowpark/relational_grouped_dataframe.py

```diff
@@ -253,6 +254,110 @@ def is_valid_tuple_for_agg(e: Union[list, tuple]) -> bool:
+                ...
+                >>> df = session.createDataFrame([('SF', 21.0), ('SF', 17.5), ('SF', 24.0), ('NY', 30.9), ('NY', 33.6)],
+                ...         schema=['location', 'temp_c'])
+                >>> df.group_by("location").apply_in_pandas(convert,
+                ...     output_schema=StructType([StructField("location", StringType()),
+                ...                               StructField("temp_c", FloatType()),
+                ...                               StructField("temp_f", FloatType())])).order_by("temp_c").show()
+                ---------------------------------------------
+                |"LOCATION"  |"TEMP_C"  |"TEMP_F"           |
+                ---------------------------------------------
+                |SF          |17.5      |63.5               |
+                |SF          |21.0      |69.8               |
+                |SF          |24.0      |75.2               |
+                |NY          |30.9      |87.61999999999999  |
+                |NY          |33.6      |92.48              |
+                ---------------------------------------------
+                <BLANKLINE>
+
+            Call ``apply_in_pandas`` using permanent UDTF with replacing original UDTF:
+
+                >>> from snowflake.snowpark.types import IntegerType, DoubleType
+                >>> _ = session.sql("create or replace temp stage mystage").collect()
+                >>> def group_sum(pdf):
+                ...     pdf.columns = ['grade', 'division', 'value']
+                ...     return pd.DataFrame([(pdf.grade.iloc[0], pdf.division.iloc[0], pdf.value.sum(), )])
+                ...
+                >>> df = session.createDataFrame([('A', 2, 11.0), ('A', 2, 13.9), ('B', 5, 5.0), ('B', 2, 12.1)],
+                ...                              schema=["grade", "division", "value"])
+                >>> df.group_by([df.grade, df.division] ).applyInPandas(
+                ...     group_sum, output_schema=StructType([StructField("grade", StringType()),
+                ...                                        StructField("division", IntegerType()),
+                ...                                        StructField("sum", DoubleType())]),
+                ...                is_permanent=True, stage_location="@mystage", name="group_sum_in_pandas", replace=True
+                ...            ).order_by("sum").show()
+                --------------------------------
+                |"GRADE"  |"DIVISION"  |"SUM"  |
+                --------------------------------
+                |B        |5           |5.0    |
+                |B        |2           |12.1   |
+                |A        |2           |24.9   |
+                --------------------------------
+                <BLANKLINE>
+
+        See Also:
+            - :class:`~snowflake.snowpark.udtf.UDTFRegistration`
+            - :func:`~snowflake.snowpark.functions.pandas_udtf`
+        """
+        import pandas as pd
```

Comment:
This may throw an exception if pandas isn't installed. How about using `from snowflake.connector.options import pandas` so user will get the same error message as in other places?

BTW, is it a must that the user's client has pandas installed?

Reply:
Pandas is required dependency here and the function description mentions this fact. Based on all examples I've seen to create vectorized UDTFs, we need pandas installed. @sfc-gh-stan, do you know if it is possible to tag vectorized UDTFs without pandas installed?